pkg/fake/ec2api.go

```diff
@@ -126,13 +128,23 @@ func (e *EC2API) CreateFleet(_ context.Context, input *ec2.CreateFleetInput, _ .
 			return nil, fmt.Errorf("missing launch template name")
 		}
 		var instanceIds []string
-		var skippedPools []CapacityPool
+		var icedPools []CapacityPool
+		var reservationExceededPools []CapacityPool
 		var spotInstanceRequestID *string
 
 		if string(input.TargetCapacitySpecification.DefaultTargetCapacityType) == karpv1.CapacityTypeSpot {
 			spotInstanceRequestID = aws.String(test.RandomName())
 		}
 
+		launchTemplates := map[string]*ec2.CreateLaunchTemplateInput{}
+		for e.CreateLaunchTemplateBehavior.CalledWithInput.Len() > 0 {
```

Comment:
I'm a bit confused what this is intending to do -- to me this is a tad odd to introspect the CalledWithInput and not the data store itself -- I'd expect we update the data store when we call CreateLaunchTemplate, not when we call CreateFleet

Reply:
I didn't realize we actually persisted our created launch templates in a `sync.Map`already - I'll update this to reference that